Add option to always get screenshots from plotter shows

AlejandroFernandezLuces opened the pull request:
### Overview

This PR adds an option to always produce screenshots instead of showing the plotter when performing a `pl.show()`. We need this feature in some PyAnsys projects, where we need to have the screenshots of several plotters without modifying the underlaying code. This is related to #5194 , which we also needed for the project.

Let me know your thoughts about this feature 🙂 


### Details

I'm using UUID library to generate unique identifiers for the screenshot filenames. This may be substituted by some attribute from PyVista, but not sure if any would ensure that the name is unique.

tkoyama010 commented on pyvista/plotting/plotter.py:
```diff
@@ -6404,7 +6405,7 @@ def __init__(
 
         self.notebook = notebook
-        if self.notebook:
+        if self.notebook or pyvista.SCREENSHOTS_ONLY:
```
This is a great feature. Thank you. I will follow the `OFF_SCREEN` naming convention and suggest the variable name `ON_SCREENSHOT`.
```suggestion
        if self.notebook or pyvista.ON_SCREENSHOT:
```

tkoyama010 reviewed (dismissed):
LGTM

AlejandroFernandezLuces commented:
Changed to pytest `tmpfolder` for image cleanup. Also, added a fixture that resets the global variables to the values it had before running the tests, I think it's cleaner and can be reused in other tests.

MatthewFlamm reviewed (approved):
I can foresee users also wanting sequentially numbered filenames to keep track of which image is which, but a future PR to add in a configurable mode.
